### ISSUE TITLE
REG: ISSUE WITH DATALODING (I WANT TO KNOW WHAT DATALODER TO USE)

### ISSUE THREAD

**Issue by Harighs** [kendrickkay/nsddatapaper#3]
**Huge problem with knowing what is right Data Loder to use for the dataset for training for deep network**

When attempting to use the dataset using AWS, I have huge problem of how to handle dataset.

**1. What data loader module to use,?
2. Is there any prebuild Data loader which returns (RGB stimuli, FMRI voxels, image captions) in specified batches.?**

Can you help me with this i am struggling for nearly 5 months.

Actual goal of my project:
I want to train transformer-based auto encoder - decoder for generating rgb features from brain data which seen the coco images.

**Comment by kendrickkay** [comment]
I am not sure what a "Data Loader" is.  Is this an AWS specific thing?  Perhaps you can look at recent papers that used NSD data and see if those papers have some associated code that you might get some insights from.

**Comment by Harighs** [comment]
> I am not sure what a "Data Loader" is. Is this an AWS specific thing? Perhaps you can look at recent papers that used NSD data and see if those papers have some associated code that you might get some insights from.

A DataLoader in the context of Python programming, especially when working with deep learning models, is essentially a tool or class that helps you manage your dataset more efficiently. 

In essence, a DataLoader simplifies the process of getting your dataset ready for training a deep learning model, making it easier to handle large datasets, work with batches, maintain data associations, and preprocess data. 

basically dataloader gives the bathces (eg. 10) of list of images eg for (10 images of rgb, 10 fmri voxel data, 10 corresponding captions of rgb image)

Do you recommend anything like this for making the trianing easy. Because i am struggling with this for nearly 5-6 months for managing the data. for my MSc Thesis

**Comment by kendrickkay** [comment]
I see.  NSD doesn't quite provide these types of loading routines for the end user, so you'll have to conjure up your own.  Alternatively, see the nsd_access (https://github.com/tknapen/nsd_access) mentioned on the first page of the NSD Data Manual
